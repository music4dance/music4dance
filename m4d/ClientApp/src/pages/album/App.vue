<script setup lang="ts">
import PageFrame from "@/components/PageFrame.vue";
import SongTable from "@/components/SongTable.vue";
import AdminFooter from "@/components/AdminFooter.vue";
import { AlbumModel } from "@/models/AlbumModel";
import { TypedJSON } from "typedjson";
import { useSongSelector } from "@/composables/useSongSelector";
import { computed } from "vue";

declare const model_: string;

const model = TypedJSON.parse(model_, AlbumModel)!;

const { songs: selected, select: selectSong } = useSongSelector();

const artistRef = computed(() => `/song/artist?name=${encodeURIComponent(model.artist || "")}`);
const hidden = computed(() => (model.artist ? ["Artist"] : []));
</script>

<template>
  <PageFrame id="app">
    <h1>
      Album: {{ model.title }}
      <span v-if="model.artist"
        >by <a :href="artistRef">{{ model.artist }}</a></span
      >
    </h1>
    <SongTable
      :histories="model.histories!"
      :filter="model.filter"
      :hide-sort="true"
      :hidden-columns="hidden"
      @song-selected="selectSong"
    ></SongTable>
    <AdminFooter :model="model" :selected="selected"></AdminFooter>
  </PageFrame>
</template>
