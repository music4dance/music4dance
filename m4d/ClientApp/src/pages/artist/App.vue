<script setup lang="ts">
import { ArtistModel } from "@/models/ArtistModel";
import { SongFilter } from "@/models/SongFilter";
import { safeDanceDatabase } from "@/helpers/DanceEnvironmentManager";
import { TypedJSON } from "typedjson";
import { Song } from "@/models/Song";
import { useSongSelector } from "@/composables/useSongSelector";

declare const model_: string;

const model = TypedJSON.parse(model_, ArtistModel)!;

const danceDB = safeDanceDatabase();
const { songs: selected, select: selectSong } = useSongSelector();

const title = `Artist: ${model.artist}`;
const filter = new SongFilter();
const histories = model.histories ?? [];
const songs = histories.map((h) => Song.fromHistory(h));
const danceIds = [
  ...songs.reduce((acc, h) => {
    const ratings = h.danceRatings;
    if (ratings) {
      ratings.forEach((r) => acc.add(r.danceId));
    }
    return acc;
  }, new Set<string>()),
];
const dances = danceIds.map((id) => danceDB.dances.find((d) => d.id === id)!);
</script>

<template>
  <PageFrame id="app" :title="title">
    <p>
      {{ histories.length }} {{ model.artist }} songs with suggestions for dances to dance to them:
      <span v-for="(dance, idx) in dances" :key="dance.id"
        ><span v-if="idx + 1 === dances.length && dances.length > 1"> and </span
        ><a :href="'/dances/' + dance.seoName">{{ dance.name }}</a
        ><span v-if="idx + 1 < dances.length && dances.length > 2">, </span></span
      >
    </p>
    <SongTable
      :histories="histories"
      :filter="filter"
      :hide-sort="true"
      :hidden-columns="['artist', 'length', 'track']"
      @song-selected="selectSong"
    ></SongTable>
    <AdminFooter :model="model" :selected="selected"></AdminFooter>
  </PageFrame>
</template>
