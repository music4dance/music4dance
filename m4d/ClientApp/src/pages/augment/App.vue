<script setup lang="ts">
import PageFrame from "@/components/PageFrame.vue";
import { AugmentModel } from "@/models/AugmentModel";
import { Song } from "@/models/Song";
import { SongDetailsModel } from "@/models/SongDetailsModel";
import { SongFilter } from "@/models/SongFilter";
import { SongHistory } from "@/models/SongHistory";
import SongCore from "@/pages/song/components/SongCore.vue";
import { TypedJSON } from "typedjson";
import AugmentInfo from "./components/AugmentInfo.vue";
import AugmentLookup from "./components/AugmentLookup.vue";
import AugmentSearch from "./components/AugmentSearch.vue";
import { getMenuContext } from "@/helpers/GetMenuContext";

import { computed, ref } from "vue";

enum AugmentPhase {
  lookup = "lookup",
  results = "results",
  edit = "edit",
}

declare const model_: AugmentModel;

const context = getMenuContext();

const model = TypedJSON.parse(model_, AugmentModel) ?? {};
const phase = ref<AugmentPhase>(AugmentPhase.lookup);
const songModel = ref<SongDetailsModel | null>(null);
const lastSong = ref<Song | null>(null);
const created = ref(false);
const propertiesString = ref("");
const tabIndex = ref(model.id ? 1 : 0);

const canAugment = computed(() => !!context.userName);

const editSong = (model: SongDetailsModel): void => {
  songModel.value = new SongDetailsModel({
    created: !!model.created,
    songHistory: model.songHistory,
    filter: new SongFilter(),
    userName: context.userName,
  });
  phase.value = AugmentPhase.edit;
};

const adminCreate = async (): Promise<void> => {
  songModel.value = new SongDetailsModel({
    created: true,
    songHistory: SongHistory.fromString(propertiesString.value),
    filter: new SongFilter(),
    userName: context.userName,
  });
  propertiesString.value = "";
  phase.value = AugmentPhase.edit;
};

const reset = (saved: boolean): void => {
  if (saved) {
    lastSong.value = Song.fromHistory((songModel.value! as SongDetailsModel).songHistory);
    created.value = !!songModel.value!.created;
  }
  phase.value = AugmentPhase.lookup;
  model.id = "";
  songModel.value = null;
};
</script>

<template>
  <PageFrame id="app">
    <BRow v-if="phase === 'lookup'">
      <BCol>
        <BAlert v-if="lastSong" dismissible show="10">
          Thank you for {{ created ? "adding" : "editing" }} <i>{{ lastSong.title }}</i> by
          {{ lastSong.artist }}
        </BAlert>
        <h1>Add Song</h1>
        <BTabs v-if="canAugment" v-model="tabIndex" card>
          <BTab title="by Title"
            ><BCardText><AugmentSearch @edit-song="editSong($event)"></AugmentSearch></BCardText
          ></BTab>
          <BTab title="by Id">
            <BCardText
              ><AugmentLookup :id="model.id!" @edit-song="editSong($event)"></AugmentLookup
            ></BCardText>
          </BTab>
          <BTab v-if="context.isAdmin" title="Admin">
            <BCardText>
              <p>Paste in text version of property list</p>
              <BInputGroup prepend="Properties">
                <BFormInput
                  id="admin-properties"
                  v-model="propertiesString"
                  palceholder="Song Properties TSV"
                  trim
                ></BFormInput>
                <BButton variant="primary" @click="adminCreate">Create</BButton>
              </BInputGroup>
            </BCardText>
          </BTab>
        </BTabs>
        <AugmentInfo v-else :id="model.id!"> </AugmentInfo>
      </BCol>
    </BRow>
    <div v-else>
      <BAlert show variant="success">
        <div v-if="songModel && songModel.created">
          <b>Create Song:</b> This song is new to music4dance, please fill in missing fields and
          click <b>Add Song</b> to add this song to the catalog. Remember, you must vote on at least
          one dance to add this song.
        </div>
        <div v-else>
          <b>Edit Song:</b> We found this song in the music4dance catalog, please vote on dance
          styles and add tags to improve the catalog.
        </div>
      </BAlert>
      <SongCore
        :model="songModel as SongDetailsModel"
        :start-editing="true"
        :creating="!!songModel && (songModel as SongDetailsModel).created"
        @song-saved="reset(true)"
        @cancel-changes="reset(false)"
      ></SongCore>
    </div>
    <BRow
      ><BCol>
        <BAlert show variant="warning">
          Adding songs is a new feature so if you run into bugs or have suggestions for improving
          this feature please don't hesitate to send email to
          <a href="mailto:info@music4dance.net">info@music4dance.net</a> or fill out our
          <a href="https://music4dance.blog/feedback/">feedback form</a> and we'll be happy to take
          your input as we contineu to improve the feature.
        </BAlert>
      </BCol>
    </BRow>
  </PageFrame>
</template>

<style lang="scss" scoped>
#service-id {
  width: 25rem;
}
</style>
