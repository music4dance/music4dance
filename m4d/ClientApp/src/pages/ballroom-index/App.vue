<script setup lang="ts">
import BallroomList from "@/components/BallroomList.vue";
import BlogTagLink from "@/components/BlogTagLink.vue";
import CompetitionCategoryTable from "@/components/CompetitionCategoryTable.vue";
import PageFrame from "@/components/PageFrame.vue";
import TempiLink from "@/components/TempiLink.vue";
import { type BreadCrumbItem, danceTrail } from "@/models/BreadCrumbItem";
import { CompetitionCategory, CompetitionGroup } from "@/models/Competition";
import { TypedJSON } from "typedjson";

declare const model_: string;
const group = TypedJSON.parse(model_, CompetitionGroup)!;
const breadcrumbs: BreadCrumbItem[] = [...danceTrail, { text: "Ballroom", active: true }];

function categoryLink(category: CompetitionCategory): string {
  return `/dances/${category.canonicalName}`;
}
</script>

<template>
  <PageFrame id="app" title="Competition Ballroom Dancing" :breadcrumbs="breadcrumbs">
    <BallroomList>
      One of the
      <a
        href="https://music4dance.blog/question-1-im-learning-to-cha-cha-where-is-some-great-music-for-practicing/"
        target="_blank"
      >
        core ideas</a
      >
      behind <a href="https://www.music4dance.net">music4dance</a> is to find interesting music to
      build playlists for competitions rounds. This page is a central location to start a search for
      songs for your own playlists based on the tempo definitions of competition ballroom dance
      established by
      <a href="http://www.worlddancesport.org/Rule/Athlete/Competition" target="_blank"
        >World Dance Council</a
      >
      and
      <a href="https://www.ndca.org/pages/ndca_rule_book/Default.asp" target="_blank"
        >National Dance Council of America</a
      >
      and our database of songs.
    </BallroomList>
    <div v-for="category in group.categories" :key="category.name">
      <h2>
        <a :href="categoryLink(category)">{{ category.name }}</a>
      </h2>
      <CompetitionCategoryTable
        :dances="category.round"
        :title="category.fullRoundTitle"
        :use-full-name="false"
      ></CompetitionCategoryTable>
    </div>
    <div>
      <TempiLink />
      <BlogTagLink title="Ballroom" tag="ballroom"></BlogTagLink>
    </div>
  </PageFrame>
</template>
