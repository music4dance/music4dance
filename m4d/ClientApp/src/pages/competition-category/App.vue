<script setup lang="ts">
import BallroomList from "@/components/BallroomList.vue";
import BlogTagLink from "@/components/BlogTagLink.vue";
import CompetitionCategoryTable from "@/components/CompetitionCategoryTable.vue";
import LinkCategory from "@/components/LinkCategory.vue";
import PageFrame from "@/components/PageFrame.vue";
import TempiLink from "@/components/TempiLink.vue";
import { ballroomTrail, type BreadCrumbItem } from "@/models/BreadCrumbItem";
import { CompetitionCategory, CompetitionGroupModel } from "@/models/Competition";
import { TypedJSON } from "typedjson";
import { computed } from "vue";

declare const model_: string;
const model = TypedJSON.parse(model_, CompetitionGroupModel)!;
const breadcrumbs: BreadCrumbItem[] = [
  ...ballroomTrail,
  { text: model?.currentCategory.name ?? "Error", active: true },
];

const category = computed(() => model?.currentCategory ?? new CompetitionCategory());

const groupTitle = computed(() => `competition ${model?.group.name.toLowerCase()} dancing`);
const groupLink = computed(
  () => `/dances/${model?.group.name.toLowerCase()}-competition-categories`,
);
</script>

<template>
  <PageFrame id="app" :title="category.name" :breadcrumbs="breadcrumbs">
    <BallroomList :name="category.name"></BallroomList>
    <CompetitionCategoryTable
      :dances="category.round"
      :title="category.fullRoundTitle"
      :use-full-name="false"
    ></CompetitionCategoryTable>
    <CompetitionCategoryTable
      v-if="category.extras && category.extras.length > 0"
      :dances="category.extras"
      :title="category.extraDancesTitle"
      :use-full-name="false"
    ></CompetitionCategoryTable>
    <div>
      Other categories of <a :href="groupLink">{{ groupTitle }}</a> are:
      <ul>
        <li v-for="cat in model.otherCategories" :key="cat.name">
          <LinkCategory :name="cat.name"></LinkCategory>
        </li>
      </ul>
    </div>
    <dl>
      <TempiLink></TempiLink>
      <BlogTagLink title="Ballroom" tag="ballroom"></BlogTagLink>
    </dl>
  </PageFrame>
</template>
