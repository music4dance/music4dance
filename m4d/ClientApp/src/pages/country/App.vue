<script setup lang="ts">
import PageFrame from "@/components/PageFrame.vue";
import { type BreadCrumbItem, danceTrail } from "@/models/BreadCrumbItem";

const breadcrumbs: BreadCrumbItem[] = [...danceTrail, { text: "Country", active: true }];
</script>

<template>
  <PageFrame id="app" title="Country Western Dance Music" :breadcrumbs="breadcrumbs">
    <p>
      Here at music4dance, we constantly seek ways to serve the broader dance community. We're
      considering adding more Country Western dances to our catalog, but we aren't experts in those
      dances. So please <a href="https://music4dance.blog/feedback/">get in touch with us</a> if
      you're a Country Western dancer and would like to see this site provide support for those
      dances in a way similar to how we support Ballroom dances, including a
      <a href="https://www.music4dance.net/dances/ballroom-competition-categories"
        >competition tempo page</a
      >, pages for each of the dances (as we currently have for
      <a href="https://www.music4dance.net/dances/country-two-step">Country Two Step</a>) and
      include those dances in the
      <a href="https://www.music4dance.net/home/counter">tempo counter</a> and
      <a href="https://www.music4dance.net/home/tempi">tempo list</a> pages.
    </p>
    <p>
      We'd especially <a href="https://music4dance.blog/feedback/">like to hear from you</a> if you
      are an expert in Country Western dances and would be willing to help us build these pages.
      We're looking to start from the tempo guidance the
      <a
        href="https://ucwdc.org/wp-content/uploads/2023/01/UCWDC-ProAm-ProPro-and-Couples-Competition-Music-BPMs-2023-2025.pdf"
        >United Country Western Dance Council</a
      >
      and the
      <a href="https://www1.worldcdf.com/pdf/RULES_BOOK.pdf">World Country Dance Federation</a>
      provide, but would be happy to have your input as well.
    </p>
  </PageFrame>
</template>
