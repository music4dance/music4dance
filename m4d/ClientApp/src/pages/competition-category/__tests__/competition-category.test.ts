import { describe, test } from "vitest";
import { testPageSnapshot } from "@/helpers/TestHelpers";
import { model } from "./model";
import App from "../App.vue";

describe("CompetitionCategory", () => {
  test("Renders a Competition Category Page", () => {
    testPageSnapshot(App, model);
  });
});
