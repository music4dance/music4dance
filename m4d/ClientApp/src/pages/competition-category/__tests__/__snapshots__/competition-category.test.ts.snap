// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CompetitionCategory > Renders a Competition Category Page 1`] = `
"<div><span context=\\"[object Object]\\">MainMenu</span>
  <nav aria-label=\\"breadcrumb\\">
    <nav aria-label=\\"breadcrumb\\" style=\\"padding: 0.25rem 0.5rem;\\">
      <ol class=\\"breadcrumb\\">
        <li class=\\"breadcrumb-item\\"><a class=\\"\\" href=\\"/\\">music4dance</a></li>
        <li class=\\"breadcrumb-item\\"><a class=\\"\\" href=\\"/dances\\">Dances</a></li>
        <li class=\\"breadcrumb-item\\"><a class=\\"\\" href=\\"/dances/ballroom-competition-categories\\">Ballroom</a></li>
        <li class=\\"breadcrumb-item active\\"><span aria-current=\\"location\\">American Smooth</span></li>
      </ol>
    </nav>
  </nav>
  <div id=\\"body-content\\" class=\\"container-fluid body-content\\">
    <h1>American Smooth</h1>
    <div>
      <div>
        <p>American Smooth is a category of <a href=\\"/dances/ballroom-competition-categories\\">competition ballroom dancing</a> that is defined by the <a href=\\"http://www.worlddancesport.org/Rule/Athlete/Competition\\">World Dance Council</a> and <a href=\\"https://www.ndca.org/pages/ndca_rule_book/Default.asp\\">National Dance Council of America</a>. </p>
        <ul style=\\"list-style-type: disc;\\">
          <li> Click on any of the tempos in the table below (in the MPM, DanceSport, or NDCA columns) to go to a list of songs of that tempo </li>
          <li>Click on the dance names to go to our details information page for that dance style</li>
          <li> Read our <a href=\\"https://music4dance.blog/im-a-competition-ballroom-dancer-can-i-find-practice-songs-that-are-a-specific-tempo/\\"> blog post</a> for step by step instructions on using our catalog to find music of specific style and tempo. </li>
          <li><a href=\\"/identity/account/register\\">Register</a> (it's free) to start building your own music lists. </li>
        </ul>
      </div>
      <div>
        <h4>An American Smooth four dance round consists of the following:</h4>
        <div class=\\"table-responsive\\" noproviderpaging=\\"false\\" noprovidersorting=\\"false\\" noproviderfiltering=\\"false\\" sortdesc=\\"false\\" selectable=\\"false\\" stickyselect=\\"false\\" selecthead=\\"true\\" selectmode=\\"multi\\" selectionvariant=\\"primary\\" busy=\\"false\\" busyloadingtext=\\"Loading...\\" perpage=\\"Infinity\\" currentpage=\\"1\\" nolocalsorting=\\"false\\" noselectonclick=\\"false\\" sortcompareoptions=\\"[object Object]\\" selecteditems=\\"\\" nosortableicon=\\"false\\" aria-busy=\\"false\\">
          <table class=\\"table b-table table-hover table-striped\\">
            <thead class=\\"\\">
              <!---->
              <tr class=\\"\\">
                <th scope=\\"col\\" class=\\"\\">Name
                  <!---->
                </th>
                <th scope=\\"col\\" class=\\"\\">MPM
                  <!---->
                </th>
                <th scope=\\"col\\" class=\\"\\">DanceSport
                  <!---->
                </th>
                <th scope=\\"col\\" class=\\"\\">NDCA
                  <!---->
                </th>
                <th scope=\\"col\\" class=\\"\\">BPM
                  <!---->
                </th>
                <th scope=\\"col\\" class=\\"\\">Meter
                  <!---->
                </th>
              </tr>
              <!---->
            </thead>
            <tbody class=\\"\\">
              <!---->
              <tr class=\\"\\">
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"slow-waltz\\">American Slow Waltz</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"/song/advancedsearch?dances=SWZ&amp;tempomin=84&amp;tempomax=90\\">28-30</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"/song/advancedsearch?dances=SWZ&amp;tempomin=84&amp;tempomax=90\\">28-30</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"/song/advancedsearch?dances=SWZ&amp;tempomin=90&amp;tempomax=90\\">30</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"/song/advancedsearch?dances=SWZ&amp;tempomin=84&amp;tempomax=90\\">84-90</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!---->3/4
                </td>
              </tr>
              <!---->
              <tr class=\\"\\">
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"tango-(ballroom)\\">American Tango (Ballroom)</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"/song/advancedsearch?dances=TGO&amp;tempomin=120&amp;tempomax=120\\">30</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"/song/advancedsearch?dances=TGO&amp;tempomin=120&amp;tempomax=120\\">30</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"/song/advancedsearch?dances=TGO&amp;tempomin=120&amp;tempomax=120\\">30</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"/song/advancedsearch?dances=TGO&amp;tempomin=120&amp;tempomax=120\\">120</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!---->4/4
                </td>
              </tr>
              <!---->
              <tr class=\\"\\">
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"slow-foxtrot\\">American Slow Foxtrot</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"/song/advancedsearch?dances=SFT&amp;tempomin=120&amp;tempomax=128\\">30-32</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"/song/advancedsearch?dances=SFT&amp;tempomin=120&amp;tempomax=128\\">30-32</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"/song/advancedsearch?dances=SFT&amp;tempomin=120&amp;tempomax=120\\">30</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"/song/advancedsearch?dances=SFT&amp;tempomin=120&amp;tempomax=128\\">120-128</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!---->4/4
                </td>
              </tr>
              <!---->
              <tr class=\\"\\">
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"viennese-waltz\\">American Viennese Waltz</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"/song/advancedsearch?dances=VWZ&amp;tempomin=159&amp;tempomax=162\\">53-54</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"/song/advancedsearch?dances=VWZ&amp;tempomin=159&amp;tempomax=162\\">53-54</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"/song/advancedsearch?dances=VWZ&amp;tempomin=159&amp;tempomax=159\\">53</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"/song/advancedsearch?dances=VWZ&amp;tempomin=159&amp;tempomax=162\\">159-162</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!---->3/4
                </td>
              </tr>
              <!---->
              <!---->
              <!---->
            </tbody>
            <!---->
            <!---->
          </table>
        </div>
      </div>
      <div>
        <h4>Other dances categorized as American Smooth, although they aren't part of the competition round, are:</h4>
        <div class=\\"table-responsive\\" noproviderpaging=\\"false\\" noprovidersorting=\\"false\\" noproviderfiltering=\\"false\\" sortdesc=\\"false\\" selectable=\\"false\\" stickyselect=\\"false\\" selecthead=\\"true\\" selectmode=\\"multi\\" selectionvariant=\\"primary\\" busy=\\"false\\" busyloadingtext=\\"Loading...\\" perpage=\\"Infinity\\" currentpage=\\"1\\" nolocalsorting=\\"false\\" noselectonclick=\\"false\\" sortcompareoptions=\\"[object Object]\\" selecteditems=\\"\\" nosortableicon=\\"false\\" aria-busy=\\"false\\">
          <table class=\\"table b-table table-hover table-striped\\">
            <thead class=\\"\\">
              <!---->
              <tr class=\\"\\">
                <th scope=\\"col\\" class=\\"\\">Name
                  <!---->
                </th>
                <th scope=\\"col\\" class=\\"\\">MPM
                  <!---->
                </th>
                <th scope=\\"col\\" class=\\"\\">DanceSport
                  <!---->
                </th>
                <th scope=\\"col\\" class=\\"\\">NDCA
                  <!---->
                </th>
                <th scope=\\"col\\" class=\\"\\">BPM
                  <!---->
                </th>
                <th scope=\\"col\\" class=\\"\\">Meter
                  <!---->
                </th>
              </tr>
              <!---->
            </thead>
            <tbody class=\\"\\">
              <!---->
              <tr class=\\"\\">
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"peabody\\">American Peabody</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"/song/advancedsearch?dances=PBD&amp;tempomin=240&amp;tempomax=240\\">60</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"/song/advancedsearch?dances=PBD&amp;tempomin=240&amp;tempomax=240\\">60</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"/song/advancedsearch?dances=PBD&amp;tempomin=240&amp;tempomax=240\\">60</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!----><a href=\\"/song/advancedsearch?dances=PBD&amp;tempomin=240&amp;tempomax=240\\">240</a>
                </td>
                <td scope=\\"col\\" class=\\"\\">
                  <!---->4/4
                </td>
              </tr>
              <!---->
              <!---->
              <!---->
            </tbody>
            <!---->
            <!---->
          </table>
        </div>
      </div>
      <div> Other categories of <a href=\\"/dances/ballroom-competition-categories\\">competition ballroom dancing</a> are: <ul>
          <li><a href=\\"/dances/international-standard\\">International Standard</a></li>
          <li><a href=\\"/dances/american-rhythm\\">American Rhythm</a></li>
          <li><a href=\\"/dances/international-latin\\">International Latin</a></li>
        </ul>
      </div>
      <dl>
        <div>
          <div><img src=\\"/images/icons/tools.png\\" alt=\\"tools\\" width=\\"24\\" height=\\"24\\"> Explore relationships between the meter and tempi of different dance styles and find songs that match.</div>
          <div class=\\"ms-3 my-1\\"><a class=\\"btn btn-sm btn-tools\\" href=\\"https://www.music4dance.net/home/tempi\\" role=\\"button\\" style=\\"margin-left: 1em; margin-bottom: 0.25em;\\"><svg viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
                <path fill=\\"currentColor\\" d=\\"M10.804 8L5 4.633v6.734zm.792-.696a.802.802 0 0 1 0 1.392l-6.363 3.692C4.713 12.69 4 12.345 4 11.692V4.308c0-.653.713-.998 1.233-.696z\\"></path>
              </svg>&nbsp;Try It!</a><a class=\\"btn btn-sm btn-tools\\" href=\\"https://music4dance.blog/music4dance-help/dance-tempi/\\" role=\\"button\\" style=\\"margin-left: 1em; margin-bottom: 0.25em;\\"><svg viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
                <g fill=\\"currentColor\\">
                  <path d=\\"M5 4a.5.5 0 0 0 0 1h6a.5.5 0 0 0 0-1zm-.5 2.5A.5.5 0 0 1 5 6h6a.5.5 0 0 1 0 1H5a.5.5 0 0 1-.5-.5M5 8a.5.5 0 0 0 0 1h6a.5.5 0 0 0 0-1zm0 2a.5.5 0 0 0 0 1h3a.5.5 0 0 0 0-1z\\"></path>
                  <path d=\\"M2 2a2 2 0 0 1 2-2h8a2 2 0 0 1 2 2v12a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2zm10-1H4a1 1 0 0 0-1 1v12a1 1 0 0 0 1 1h8a1 1 0 0 0 1-1V2a1 1 0 0 0-1-1\\"></path>
                </g>
              </svg>&nbsp;Documentation</a><a class=\\"btn btn-sm btn-tools\\" href=\\"https://music4dance.blog/tag/tempo/\\" role=\\"button\\" style=\\"margin-left: 1em; margin-bottom: 0.25em;\\"><svg viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
                <g fill=\\"currentColor\\">
                  <path d=\\"M15.502 1.94a.5.5 0 0 1 0 .706L14.459 3.69l-2-2L13.502.646a.5.5 0 0 1 .707 0l1.293 1.293zm-1.75 2.456l-2-2L4.939 9.21a.5.5 0 0 0-.121.196l-.805 2.414a.25.25 0 0 0 .316.316l2.414-.805a.5.5 0 0 0 .196-.12l6.813-6.814z\\"></path>
                  <path fill-rule=\\"evenodd\\" d=\\"M1 13.5A1.5 1.5 0 0 0 2.5 15h11a1.5 1.5 0 0 0 1.5-1.5v-6a.5.5 0 0 0-1 0v6a.5.5 0 0 1-.5.5h-11a.5.5 0 0 1-.5-.5v-11a.5.5 0 0 1 .5-.5H9a.5.5 0 0 0 0-1H2.5A1.5 1.5 0 0 0 1 2.5z\\"></path>
                </g>
              </svg>&nbsp;Blog Posts</a></div>
          <!--v-if-->
        </div>
        <div>
          <dd><img src=\\"/vclient/src/assets/images/icons/info.png\\" alt=\\"tools\\" width=\\"24\\" height=\\"24\\"> Read the <a href=\\"https://music4dance.blog\\">music4dance blog</a> entries about <a href=\\"https://music4dance.blog/tag/ballroom\\">Ballroom</a>. </dd>
        </div>
      </dl>
    </div>
  </div>
  <div id=\\"footer-content\\">
    <hr>
    <footer>
      <p> © 2024 - <a href=\\"https://www.music4dance.net\\">Music4Dance.net</a> - <a href=\\"https://www.music4dance.net/home/sitemap\\">Site Map</a> - <a href=\\"https://www.music4dance.net/home/termsofservice\\">Terms of Service</a> - <a href=\\"https://www.music4dance.net/home/privacypolicy\\">Privacy Policy</a> - <a href=\\"https://www.music4dance.net/home/credits\\">Credits</a> - <a>Help</a></p>
    </footer>
  </div>
</div>"
`;
